utils/apiclient.py

```diff
@@ -20,10 +20,38 @@ def send_rpc(self, method: str, params: tp.Optional[tp.Any] = None) -> tp.Dict:
             if not isinstance(params, (list, tuple)):
                 params = [params]
             body["params"] = params
+            
         resp = self.post(self.proxy_url, json=body, timeout=60)
         response_body = resp.json()
         if "result" not in response_body and "error" not in response_body:
-            raise AssertionError("Request must contains 'result' or 'error' field")
+            raise AssertionError(
+                "Request must contains 'result' or 'error' field")
+
+        if "error" in response_body:
+            assert "result" not in response_body, "Response can't contains error and result"
+        if "error" not in response_body:
+            assert response_body["id"] == req_id
+
+        return response_body
+
+    def tracer_send_rpc(self, method: str, req_type: str, params: tp.Optional[tp.Any] = None) -> tp.Dict:
```

Comment:
It looks confusing: JsonRPCSession class have only proxy_url field but tracer_send_rpc and send_rpc methods
I think it is better to create 2 different classes or create one JsonRPCSession as a parent

Reply:
rewrote JsonRPCSession class to be compatible with proxy and tracer rpc calls